assembl/graphql/schema.py

```diff
@@ -1,81 +1,439 @@
+        identifier = graphene.String(required=True)
+        video = graphene.Argument(VideoInput)
+        questions = graphene.List(QuestionInput)
+        # TODO upload img example http://docs.pylonsproject.org/projects/pyramid-cookbook/en/latest/forms/file_uploads.html
+
+    thematic = graphene.Field(lambda: Thematic)
+
+    @staticmethod
+    def mutate(root, args, context, info):
+        cls = models.Thematic
+        discussion_id = context.matchdict['discussion_id']
+        user_id = context.authenticated_userid or Everyone
+
+        permissions = get_permissions(user_id, discussion_id)
+        allowed = cls.user_can_cls(user_id, CrudPermissions.CREATE, permissions)
+        if not allowed or (allowed == IF_OWNED and user_id == Everyone):
+            raise HTTPUnauthorized()
+
+        identifier = args.get('identifier')
+        with cls.default_db.no_autoflush:
+            title_entries = args.get('title_entries')
+            if len(title_entries) == 0:
+                raise Exception('Thematic titleEntries needs at least one entry')
+                # Better to have this message than
+                # 'NoneType' object has no attribute 'owner_object'
+                # when creating the saobj below if title=None
+
+            title_langstring = langstring_from_input_entries(title_entries)
+            description_langstring = langstring_from_input_entries(args.get('description_entries'))
+            kwargs = {}
+            if description_langstring is not None:
+                kwargs['description'] = description_langstring
+
+            video = args.get('video')
+            if video is not None:
+                video_title = langstring_from_input_entries(video['title_entries'])
+                if video_title is not None:
+                    kwargs['video_title'] = video_title
+
+                video_description = langstring_from_input_entries(video['description_entries'])
+                if video_description is not None:
+                    kwargs['video_description'] = video_description
+
+                kwargs['video_html_code'] = video['html_code']
+
+            saobj = cls(
+                discussion_id=discussion_id,
+                title=title_langstring,
+                identifier=identifier,
+                **kwargs)
+            db = saobj.db
+            db.add(saobj)
+            db.flush()
+
+            questions_input = args.get('questions')
+            if questions_input is not None:
+                for question_input in questions_input:
+                    title_ls = langstring_from_input_entries(
+                        question_input['title_entries'])
+                    saobj.children.append(
+                        models.Question(
+                            title=title_ls,
+                            discussion_id=discussion_id
+                        )
+                    )
+                db.flush()
+
+        return CreateThematic(thematic=saobj)
```

Comment:
AHhhh okay. I lost you because you're returning the `Mutation` class with the object identified. It does make sense looking back up to what GraphQL does.